src/test/java/com/amazon/ion/conformance/expectations.kt

```diff
@@ -160,8 +162,8 @@ private fun TestCaseSupport.denotesModelContent(modelContent: AnyElement, reader
 }
 
 private fun TestCaseSupport.denotesNull(expectation: SeqElement, reader: IonReader) {
-    val expectedType = expectation.tail.single().textValue.uppercase().let(IonType::valueOf)
-    val actualType = reader.next()
+    val expectedType = expectation.tail.singleOrNull()?.textValue?.uppercase()?.let(IonType::valueOf) ?: IonType.NULL
+    val actualType = reader.type
```

Comment:
🗺️ Two issues here.
First, the type for the "denotes Null" clause is optional. Second, calling `next()` here instead of `type` meant that it would skip over every null value and then see if the next value is null.